anchore_engine/services/policy_engine/engine/vulns/mappers.py

```diff
@@ -157,180 +524,29 @@ def transform_matches_to_vulnerabilities(self, grype_response):
-            if fix_ver:
-                fix_observed_at = now
-            else:
-                fix_ver = "None"  # barf
-                fix_observed_at = None
-
-            engine_nvd_scores = []
-            engine_vendor_scores = []  # TODO replace with grype data when available
-
-            grype_cvss_v2 = vuln.get("cvssV2")
-            grype_cvss_v3 = vuln.get("cvssV3")
-
-            if grype_cvss_v2 or grype_cvss_v3:
-                engine_score = CvssCombined(id=vuln_id)
-                engine_nvd_scores.append(engine_score)
-
-                if grype_cvss_v2:
-                    engine_score.cvss_v2 = CvssScore(
-                        base_score=grype_cvss_v2.get("baseScore"),
-                        exploitability_score=grype_cvss_v2.get("exploitabilityScore"),
-                        impact_score=grype_cvss_v2.get("impactScore"),
-                    )
-
-                if grype_cvss_v3:
-                    engine_score.cvss_v3 = CvssScore(
-                        base_score=grype_cvss_v3.get("baseScore"),
-                        exploitability_score=grype_cvss_v3.get("exploitabilityScore"),
-                        impact_score=grype_cvss_v3.get("impactScore"),
-                    )
-
-            pkg_type = self.__grype_engine_pkg_type_map__.get(artifact.get("type"))
-            pkg_name = artifact.get("name")
-            pkg_version = artifact.get("version")
-            pkg_path = (
-                artifact.get("locations")[0].get("path")
-                if artifact.get("locations")
-                else "NA"
-            )
-
-            search_key = item.get("matchDetails").get("searchKey")
-
-            # TODO replace with grype data when available
-            is_legacy = False
-            if "distro" in search_key:
-                is_legacy = True
-                feed = "vulnerabilities"
-                distro = search_key.get("distro")
-
-                distro_type = distro.get("type")
-                if distro_type.lower() == "redhat":
-                    feed_group = "rhel:"
-                else:
-                    feed_group = distro_type.lower() + ":"
-
-                distro_version = distro.get("version")
-                group = distro_version.split(".", 1)[0]
-                feed_group += group
-            elif vuln_id.startswith("GHSA"):
-                is_legacy = True
-                feed = "vulnerabilities"
-                feed_group = "github:{}".format(pkg_type)
-            elif vuln_id.startswith("VULNDB"):  # this is absolutely WRONG!!!
-                feed = "vulndb"
-                feed_group = "vulndb:vulnerabilities"
-            else:
-                feed = "nvdv2"
-                feed_group = "nvdv2:cves"
-
-            vuln_tuple = (vuln_id, feed_group, pkg_name, pkg_version, pkg_path)
-            if vuln_tuple in unique_results:
+            pkg_mapper_class = GRYPE_PACKAGE_MAPPERS.get(artifact.get("type"))
```

Comment:
Will artifact ever be `None`? If so the get on line 532 could throw an error

Reply:
If an attribute is not present in a dictionary, `.get()` will simply return `None` Further ops on null dereferenced variable will raise errors. And yes, I do agree this needs a lot of unit testing and it's still waiting on grype updates